components/brave_shields/browser/filter_list_catalog_entry.cc

```diff
@@ -70,6 +70,20 @@ bool GetStringVector(const base::Value* value,
   }
 }
 
+bool GetUint8(const base::Value* value, uint8_t* field) {
+  DCHECK(field);
+  if (value == nullptr || !value->is_int()) {
+    return false;
+  } else {
+    int i = value->GetInt();
+    if (i > UINT8_MAX) {
+      return false;
+    }
+    *field = (uint8_t)i;
```

Comment:
I updated the check above to the following, although it doesn't appear in the preview here for some reason:
```cpp
if (i < 0 || i > UINT8_MAX) {
  return false;
}
```

Reply:
@bridiver should we use some native helper here?

Reply:
There's `base::IsValueInRangeForNumericType<uint8_t>(i)` but presumedly that won't help with the cast warning either. Closest I found for a one-liner was `base::checked_cast<uint8_t>(value->GetInt())` but that CHECKs on out-of-range values instead of returning.

Reply:
we definitely should not be doing a c-style cast anywhere, but `base::checked_cast` is probably what you want here. If you want to fail without a CHECK then call `IsValueInRangeForNumericType` first

Reply:
https://chromium.googlesource.com/chromium/src/+/main/base/numerics/README.md